packages/appsync-emulator-serverless/serverCore.js

```diff
@@ -79,18 +112,17 @@ class SubscriptionServer {
       });
     }
 
-    const { asyncIterator, topicId } = reg;
-    log.info('clientConnect', { clientId, topicId });
+    const { asyncIterator, topicId, variables } = reg;
+    log.info('clientConnect', { clientId, topicId, variables });
 
     // eslint-disable-next-line no-constant-condition
     while (true) {
       const { value: payload, done } = await asyncIterator.next();
       if (done) break;
-      if (
-        payload == null ||
-        (typeof payload === 'object' && payload.data == null)
-      ) {
-        log.info('subscribe payload is null skipping publish', payload);
+
+      const shouldPublish = shouldPublishSubscription(payload, variables);
+
+      if (!shouldPublish) {
```

Comment:
very minor nit I would avoid assigning _shouldPublish* only to use it in the if statement.